toast-tk-runtime/src/main/java/io/toast/tk/runtime/parse/TestParser.java

```diff
@@ -28,37 +27,31 @@ public TestParser() {
 		LOG.info("Parser intializing..");
 	}
 
-	public ITestPage parse(String path) throws IOException, IllegalArgumentException {
-		path = cleanPath(path);
-		final Path p = Paths.get(path);
-		List<String> list;
-		try(Stream<String> lines = Files.lines(p)) {
-			list = lines.collect(Collectors.toList());
+	public ITestPage parse(List<String> lines, String filename) throws IOException, IllegalArgumentException {
+		if (lines.isEmpty()) {
+			throw new IllegalArgumentException("File " + filename + " is empty");
 		}
-		assertFileWasNotEmpty(path, list);
-		removeBom(list);
-		return buildTestPage(list, p.getFileName().toString(), path);
+		return buildTestPage(lines, filename);
 	}
 
-	private static void assertFileWasNotEmpty(
-		final String path, 
-		final List<String> list
-	) {
-		if (list.isEmpty()) {
-			throw new IllegalArgumentException("File empty at path: " + path);
+	private List<String> getStrings(InputStream inputStream) throws IOException {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private method 'getStrings' is never used. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.talanlabs.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)